unity-renderer/Assets/DCLPlugins/ECS7/Systems/PlayerSystem/ECSPlayerTransformSystem.cs

```diff
@@ -11,42 +11,49 @@ namespace ECSSystems.PlayerSystem
 {
     public class ECSPlayerTransformSystem : IDisposable
     {
+        private readonly BaseList<IParcelScene> loadedScenes;
         private readonly BaseVariable<Transform> avatarTransform;
-        private readonly RendererState rendererState;
         private readonly Vector3Variable worldOffset;
-        private readonly BaseList<IParcelScene> loadedScenes;
         private readonly IECSComponentWriter componentsWriter;
 
+        private Transform currentAvatarTransform = null;
+        private Vector3 currentWorldOffset = Vector3.zero;
+
         private Vector3 lastAvatarPosition = Vector3.zero;
         private Quaternion lastAvatarRotation = Quaternion.identity;
         private long timeStamp = 0;
         private bool newSceneAdded = false;
 
-        public ECSPlayerTransformSystem(IECSComponentWriter componentsWriter)
+        public ECSPlayerTransformSystem(IECSComponentWriter componentsWriter,
+            BaseList<IParcelScene> sdk7Scenes, BaseVariable<Transform> avatarTransform, Vector3Variable worldOffset)
         {
-            avatarTransform = DataStore.i.world.avatarTransform;
-            rendererState = CommonScriptableObjects.rendererState;
-            worldOffset = CommonScriptableObjects.worldOffset;
-            loadedScenes = DataStore.i.ecs7.scenes;
+            loadedScenes = sdk7Scenes;
+            this.avatarTransform = avatarTransform;
+            this.worldOffset = worldOffset;
             this.componentsWriter = componentsWriter;
 
             loadedScenes.OnAdded += OnSceneLoaded;
+            avatarTransform.OnChange += OnAvatarTransformChanged;
```

Comment:
Subscriptions look redundant: you can access `avatarTransform` and `worldOffset` directly.
Why did you decide to introduce subscriptions?

Reply:
I still don't agree, though I can understand that accessing property may be more expensive, it looks like a bad pattern: and create a precedent: subscribing and saving to a local variable vs direct access to a property. 
I ping @AjimenezDCL, @Kinerius for a second opinion

Reply:
benchmarking this in an isolated way will not be noticeable, 
but we can't keep disregarding small performance improvements just due lack of pragmatism 
because at the end every small performance improvement will all add up as a whole.

@mikhail-dcl I did those changes here https://github.com/decentraland/unity-renderer/pull/3975/commits/1963d357ff3cf3f1c07d23bde5f5c350ee82bbe3 since this PR is blocking PRs in other 4 repositories and I need this merged asap